### PR TITLE
DSND-2339: implement field formatting functions

### PR DESCRIPTION
**Pull request by crowlandsCH**
### Related Jira tickets
[DSND-2339](https://companieshouse.atlassian.net/browse/DSND-2339)

### General
- [ ] Is the PR as small as it can be?
- [ ] Has the Jira ticket been updated with any relevant comments?
- [ ] Is the `README` up to date?
- [ ] Has the `POM` been updated for the latest versions of dependencies?
- [ ] Has the code been double-checked against `main`?
- [ ] Does the code adhere standards in this repository, including SonarQube checks?

### Testing
- [ ] Do the code changes have unit and integration tests with code coverage > 80%?
- [ ] Has the code been tested locally and/or passed the API Karate tests on Tilt?
- [ ] Have mandatory manual test cases been run?
- [ ] Are extra Karate tests required?
- [ ] Are all the test data resources up to date?

### Release preparation
_Where possible, add links to the respective Jira tickets when an item is checked_
- [ ] Have these changes been included in a release ticket?
- [ ] Are changes required to the `docker-chs-development` deployment or test data?
- [ ] Are any changes required to the environment added to `chs-configs`?

## Notes to the tester
_Add any testing hints or instructions here._

[DSND-2339]: https://companieshouse.atlassian.net/browse/DSND-2339?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by johncook-ch — CHANGES_REQUESTED** [review]

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/LowerCase.java [hunk 1]
```diff
@@ -3,29 +3,35 @@
...
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.List;
+import java.util.Locale;
 import java.util.Map;
-import uk.gov.companieshouse.filinghistory.consumer.delta.transformrules.rules.SetterArgs;
-
+import java.util.regex.Pattern;
+import org.eclipse.jetty.util.StringUtil;
+import org.springframework.stereotype.Component;
+@Component
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/LowerCase.java:11]: done

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/BsonDate.java [hunk 2]
```diff
@@ -3,9 +3,11 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.List;
 import java.util.Map;
-import uk.gov.companieshouse.filinghistory.consumer.delta.transformrules.rules.SetterArgs;
-
+import org.eclipse.jetty.util.StringUtil;
+import org.springframework.stereotype.Component;
+@Component
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/BsonDate.java:16]: done

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/AddressCase.java [hunk 3]
```diff
@@ -3,9 +3,10 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.List;
 import java.util.Map;
-import uk.gov.companieshouse.filinghistory.consumer.delta.transformrules.rules.SetterArgs;
-
+import org.springframework.stereotype.Component;
+@Component
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/AddressCase.java:13]: done

File: src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/BsonDate.java [hunk 4]
```diff
@@ -3,9 +3,11 @@
 import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.fasterxml.jackson.databind.node.ObjectNode;
+import java.util.List;
 import java.util.Map;
-import uk.gov.companieshouse.filinghistory.consumer.delta.transformrules.rules.SetterArgs;
-
+import org.eclipse.jetty.util.StringUtil;
```
[the remark this answers is not in the document]
Comment (crowlandsCH) [reply src/main/java/uk/gov/companieshouse/filinghistory/consumer/delta/transformrules/functions/BsonDate.java]: using apache stringutils everywhere for now

**Comment by ch-code-analysis** [comment]
CI: No security warnings found

**Review by johncook-ch — APPROVED** [review]

**Review by cgregoryCH — APPROVED** [review]

**Pull request merged**